PasswordStorage.php

```diff
@@ -19,10 +19,25 @@ class PasswordStorage {
     const HASH_SALT_INDEX = 3;
     const HASH_PBKDF2_INDEX = 4;
 
+    /**
+     * Hash a password with PBKDF2
+     * 
+     * @param string $password
+     * @return string
+     */
     public static function create_hash($password)
     {
         // format: algorithm:iterations:outputSize:salt:pbkdf2output
-        $salt_raw = mcrypt_create_iv(self::PBKDF2_SALT_BYTES, MCRYPT_DEV_URANDOM);
+        if (!\is_string($password)) {
+            throw new InvalidArgumentException(
+                "create_hash(): Expected a string"
+            );
+        }
+        if (\function_exists('random_bytes')) {
+            $salt_raw = \random_bytes(self::PBKDF2_SALT_BYTES);
+        } else {
+            $salt_raw = \mcrypt_create_iv(self::PBKDF2_SALT_BYTES, MCRYPT_DEV_URANDOM);
+        }
         if ($salt_raw === false) {
```

Comment:
`random_bytes` doesn't return `false` on error, it throws exceptions. We kinda promise to transform all exceptions into one of our two exception types (which is kinda stupid now that I think of it). So we have to either give up on that promise or put a try-catch around `random_bytes` that'll convert it into a `CannotPerformOperationException`.